src/common/workspace-store.ts

```diff
@@ -67,14 +71,20 @@ export class WorkspaceStore extends BaseStore<WorkspaceStoreModel> {
   }
 
   @action
-  saveWorkspace(workspace: Workspace) {
-    const id = workspace.id;
+  saveWorkspace(workspace: Workspace): { workspace?: Workspace, error?: string} {
```

Comment:
I know it is slightly more work but wouldn't it be better to have this return value be `{ workspace: Workspace } | { error: string }`?

Reply:
@Nokel81 I guess in this case it would be a bit harder to consume results of this function in typescript. 
@aleksfront wdyt?

Reply:
A little (you have to do `"workspace" in res` but I think that it better communicates what the intent is.

Reply:
What if this function will return a value **only** if it has error? So return type would be as simple as `string`. If saving goes okay, it won't return anything.

Reply:
Makes sense, `string | null` since I see the only case where this function's return value is being used is to check the error.